pkg/utils/regexutils/regex.go

```
@@ -40,3 +43,23 @@ func NewRegexWithProgramSize(regex string, programsize *uint32) *envoy_type_matc
 		Regex: regex,
 	}
 }
+
+func ConvertRegexMatchAndSubstitute(params plugins.RouteParams, in *v32.RegexMatchAndSubstitute) *envoy_type_matcher_v3.RegexMatchAndSubstitute {
+	if in == nil {
+		return nil
+	}
+
+	out := &envoy_type_matcher_v3.RegexMatchAndSubstitute{
+		Pattern:      NewRegex(params.Ctx, in.GetPattern().GetRegex()),
+		Substitution: in.GetSubstitution(),
+	}
+	switch inET := in.GetPattern().GetEngineType().(type) {
+	case *v32.RegexMatcher_GoogleRe2:
+		outET := out.GetPattern().GetEngineType().(*envoy_type_matcher_v3.RegexMatcher_GoogleRe2)
+		if inET.GoogleRe2.GetMaxProgramSize() != nil && (outET.GoogleRe2.GetMaxProgramSize() == nil || inET.GoogleRe2.GetMaxProgramSize().GetValue() < outET.GoogleRe2.GetMaxProgramSize().GetValue()) {
+			out.Pattern = NewRegexWithProgramSize(in.GetPattern().GetRegex(), &inET.GoogleRe2.GetMaxProgramSize().Value)
+		}
+	}
```

let's add a `default` case ere that returns an error if this is an unknown engine type. Just so that if something changes in the API, this won't fail silently.

---

Ah I see this was copied over from basicroute/plugin.go. This is a nit then.

---

Added a default which outputs an error. 
It cause the unit test to fail (which I've updated to now provide a value value).